hazelcast/src/main/java/com/hazelcast/sql/SqlResult.java

```diff
@@ -35,13 +41,32 @@
  * }
  * </pre>
  *
- * @see #iterator()
- * @see #close()
+ * <p>
+ * <h4>Usage for update count</h4>
+ *
+ * <pre>
+ *     long updated = hazelcastInstance.getSql().query("UPDATE ...");
+ * </pre>
+ *
+ * You don't need to call {@link #close()} in this case.
  */
 public interface SqlResult extends Iterable<SqlRow>, AutoCloseable {
+
+    /**
+     * If this result represents a row set, this method returns {@code false}.
+     * If this result represents an update count (such as for a DML query), it
+     * returns {@code true}.
+     *
+     * @return {@code false} for a rows result and {@code true} for an update
+     *     count result
+     */
+    boolean isUpdateCount();
```

Comment:
A minor thought to consider - we may have a method to check for "update count", like now, or we may have a method to check for result set, e.g. `hasResultSet`. Which one is better?

Reply:
I liked the `isUpdateCount` better, but would find `hasResultSet` just as fine...